lib/classes/Variables.js

```diff
@@ -683,32 +683,28 @@ class Variables {
       // eslint-disable-next-line global-require, import/no-dynamic-require
       const jsFile = require(referencedFileFullPath);
       const variableArray = variableString.split(':');
-      let returnValueFunction;
+      let returnValue;
       if (variableArray[1]) {
         let jsModule = variableArray[1];
         jsModule = jsModule.split('.')[0];
-        returnValueFunction = jsFile[jsModule];
+        returnValue = jsFile[jsModule];
       } else {
-        returnValueFunction = jsFile;
+        returnValue = jsFile;
       }
 
-      if (typeof returnValueFunction !== 'function') {
-        const errorMessage = [
-          'Invalid variable syntax when referencing',
-          ` file "${referencedFileRelativePath}".`,
-          ' Check if your javascript is exporting a function that returns a value.',
-        ].join('');
-        return BbPromise.reject(new this.serverless.classes.Error(errorMessage));
+      if (typeof returnValue === 'function') {
+        valueToPopulate = returnValue.call(jsFile, this.serverless);
+      } else {
+        valueToPopulate = returnValue;
       }
-      valueToPopulate = returnValueFunction.call(jsFile, this.serverless);
 
       return BbPromise.resolve(valueToPopulate).then(valueToPopulateResolved => {
         let deepProperties = variableString.replace(matchedFileRefString, '');
         deepProperties = deepProperties.slice(1).split('.');
         deepProperties.splice(0, 1);
         return this.getDeeperValue(deepProperties, valueToPopulateResolved).then(
           deepValueToPopulateResolved => {
-            if (typeof deepValueToPopulateResolved === 'undefined') {
+            if (['undefined', 'symbol', 'function'].includes(typeof deepValueToPopulateResolved)) {
```

Comment:
Let's seclude that to maybe `nonJsonTypes` _set_ in module body, and validate via:

```javacript
if (nonJsonTypes.has(typeof deepValueToPopulateResolved)) {
```